ddtrace/_monkey.py

```diff
@@ -210,98 +218,35 @@ def patch(raise_errors=True, patch_modules_prefix=DEFAULT_MODULES_PREFIX, **patc
 
         >>> patch(psycopg=True, elasticsearch=True)
     """
-    modules = [m for (m, should_patch) in patch_modules.items() if should_patch]
-    for module in modules:
-        if module in _PATCH_ON_IMPORT:
-            modules_to_poi = _PATCH_ON_IMPORT[module]
-            for m in modules_to_poi:
-                # If the module has already been imported then patch immediately
-                if m in sys.modules:
-                    _patch_module(module, patch_modules_prefix=patch_modules_prefix, raise_errors=raise_errors)
-                    break
-                # Otherwise, add a hook to patch when it is imported for the first time
-                else:
-                    # Use factory to create handler to close over `module` and `raise_errors` values from this loop
-                    when_imported(m)(_on_import_factory(module, raise_errors))
-
-            # manually add module to patched modules
-            with _LOCK:
-                _PATCHED_MODULES.add(module)
-        else:
-            _patch_module(module, patch_modules_prefix=patch_modules_prefix, raise_errors=raise_errors)
+    contribs = [c for c, should_patch in patch_modules.items() if should_patch]
+    for contrib in contribs:
+        # Check if we have the requested contrib.
+        if not os.path.isfile(os.path.join(os.path.dirname(__file__), "contrib", contrib, "__init__.py")):
```

Comment:
🤔 I think that was a different error for when we couldn't find the `patch` function. Here we raise `ModuleNotFoundException` instead because we failed to find the contrib module.

Reply:
Okay, then maybe we can remove the `IntegrationNotAvailableException` class since that is no longer being used.